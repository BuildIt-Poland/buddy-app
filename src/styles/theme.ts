import { createMuiTheme, responsiveFontSizes } from '@material-ui/core/styles';
import { colorSchemes } from '@buildit/gravity-particles';

const { tealWhite } = colorSchemes;

let theme = createMuiTheme({
  typography: {
    htmlFontSize: 10, //62,5%
    button: {
      textTransform: 'none',
    },
    fontFamily:
      'Wipro Akkurat TT, Wipro Akkurat Mono TT, Roboto, Helvetica, Arial, sans-serif',
    h1: {
      fontSize: '3rem',
    },
    h2: {
      fontSize: '2.4rem',
    },
    h3: {
      fontSize: '2rem',
    },
    h4: {
      fontSize: '1.8rem',
    },
    h5: {
      fontSize: '1.6rem',
    },
    subtitle1: {
      fontSize: '1.4rem',
    },
    subtitle2: {
      fontSize: '1.2rem',
    },
  },
  palette: {
    type: 'light',
    primary: {
      main: tealWhite.groupB.accent,
      contrastText: tealWhite.groupA.neutral,
    },
    secondary: {
      main: tealWhite.groupB.neutral,
      contrastText: tealWhite.groupA.neutral,
    },
    background: {
      default: tealWhite.groupA.neutral,
    },
    text: {
      primary: tealWhite.groupB.neutral,
    },
  },
});

theme = responsiveFontSizes(theme);

export default theme;
