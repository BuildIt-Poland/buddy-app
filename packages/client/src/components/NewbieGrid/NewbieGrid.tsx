import React from 'react';
import { Link } from 'react-router-dom';
import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import { makeStyles, Theme } from '@material-ui/core/styles';
import { Newbie } from '@buddy-app/schema';
import { ROUTES } from 'shared/routes';
import Avatar from 'components/Avatar';
import useTaskProgress from 'hooks/useTaskProgress';
import { getProgressInPercentages } from 'utils';

const useStyles = makeStyles<Theme>(theme => ({
  card: {
    height: '25rem',
    width: '25rem',
    padding: theme.spacing(2),
    borderRadius: '15%',
    textDecoration: 'none',
    background: `linear-gradient(${theme.palette.background.default}, ${theme.palette.background.paper})`,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    transition: '0.5s all ease',
    transform: 'rotateX(5deg) scale(0.9)',
    boxShadow: theme.shadows[5],
    '&:hover': {
      boxShadow: theme.shadows[7],
      transform: 'rotate(0deg) scale(1.05)',
    },
  },
  cardInfo: {
    transform: 'scale(1.2)',
  },
  gridItem: {
    perspective: '900px',
    padding: theme.spacing(1),
    [theme.breakpoints.down('xs')]: {
      padding: theme.spacing(0.5),
    },
  },
  grid: {
    [theme.breakpoints.down('xs')]: {
      paddingTop: theme.spacing(2),
      overflowX: 'auto',
    },
    listStyleType: 'none',
    margin: 0,
    padding: 0,
  },
  link: {
    textDecoration: 'none',
  },
}));

const NewbieGrid: React.FC<{ newbies: Newbie[] }> = ({ newbies }) => {
  const classes = useStyles();
  const { getBuddyProgress } = useTaskProgress();

  return (
    <Grid
      component='ul'
      container
      wrap={'wrap'}
      justify='center'
      className={classes.grid}>
      {newbies.map(newbie => (
        <Grid component='li' key={newbie.id} item className={classes.gridItem}>
          <Link
            className={classes.link}
            to={ROUTES.BUDDY_TASKS_LIST.replace(':newbieId', newbie.id)}>
            <Card className={classes.card}>
              <CardContent className={classes.cardInfo}>
                <Avatar
                  progress={getProgressInPercentages(getBuddyProgress(newbie))}
                  name={newbie.name}
                  imgSrc={newbie.photo}
                  position={newbie.position || undefined}
                />
              </CardContent>
            </Card>
          </Link>
        </Grid>
      ))}
    </Grid>
  );
};

export default NewbieGrid;
