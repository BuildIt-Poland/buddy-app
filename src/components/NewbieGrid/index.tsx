import NewbieGrid from './NewbieGrid';

export default NewbieGrid;
