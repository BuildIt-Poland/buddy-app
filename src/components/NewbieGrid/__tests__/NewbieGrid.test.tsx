import React from 'react';
import { create } from 'react-test-renderer';
import ThemeProvider from '@material-ui/styles/ThemeProvider';
import { MemoryRouter } from 'react-router';
import theme from 'styles/theme';
import NewbieGrid from '../NewbieGrid';

jest.mock('@material-ui/core/Grid', () => 'Grid');
jest.mock('@material-ui/core/Card', () => 'Card');
jest.mock('@material-ui/core/CardContent', () => 'CardContent');
jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('components/Avatar', () => 'Avatar');

describe('Component - NewbieGrid', () => {
  describe('renders correctly', () => {
    const newbies = [
      {
        id: 'ck17svulh9k2k0b17j31ansfk',
        photo: null,
        name: 'Test 2',
        startDate: null,
        tasksInfo: { buddyProgress: 0.5 },
      },
      {
        id: 'ck17swp2m9kcv0b17we0ibrdn',
        photo: null,
        name: 'Test 1',
        startDate: '2019-01-01',
        position: 'Front End',
        tasksInfo: { buddyProgress: 0.6 },
      },
    ];

    test('Should render correctly NewbieGrid', () => {
      const component = create(
        <MemoryRouter>
          <ThemeProvider theme={theme}>
            <NewbieGrid newbies={newbies} />
          </ThemeProvider>
        </MemoryRouter>
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
