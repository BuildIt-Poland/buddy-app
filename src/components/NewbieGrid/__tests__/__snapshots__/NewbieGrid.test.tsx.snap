// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Carrousel renders correctly Should render correctly Carrousel 1`] = `
<Grid
  className="makeStyles-carrousel-4"
  container={true}
  justify="center"
  wrap="wrap"
>
  <Grid
    className="makeStyles-carrouselItem-3"
    item={true}
  >
    <a
      className="makeStyles-link-5"
      href="/buddy/newbies/ck17svulh9k2k0b17j31ansfk/tasks"
      onClick={[Function]}
    >
      <Paper
        className="makeStyles-carrouselElement-1"
      >
        <Box
          className="makeStyles-carrouselElementInfo-2"
        >
          <Avatar
            imgSrc={null}
            name="Test 2"
            progress={50}
          />
        </Box>
      </Paper>
    </a>
  </Grid>
  <Grid
    className="makeStyles-carrouselItem-3"
    item={true}
  >
    <a
      className="makeStyles-link-5"
      href="/buddy/newbies/ck17swp2m9kcv0b17we0ibrdn/tasks"
      onClick={[Function]}
    >
      <Paper
        className="makeStyles-carrouselElement-1"
      >
        <Box
          className="makeStyles-carrouselElementInfo-2"
        >
          <Avatar
            imgSrc={null}
            name="Test 1"
            position="Front End"
            progress={60}
          />
        </Box>
      </Paper>
    </a>
  </Grid>
</Grid>
`;
